heartbeat/monitors/wrappers/monitorstate/monitorstate.go

```diff
@@ -124,14 +126,16 @@ func (s *State) recordCheck(sf stdfields.StdMonitorFields, newStatus StateStatus
 			}
 		}
 
-		if !hasStabilized { // continue flapping
+		if !hasStabilized || !isFinalAttempt { // continue flapping
```

Comment:
We still don't have flapping turned on, but I thought I'd at least adapt it to work right here